overlord/configstate/configcore/watchdog.go

```diff
@@ -0,0 +1,134 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+var watchdogConfigKeys = map[string]bool{
+	"RuntimeWatchdogSec":  true,
+	"ShutdownWatchdogSec": true,
+}
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.watchdog.runtime-timeout"] = true
+	supportedConfigurations["core.watchdog.shutdown-timeout"] = true
+}
+
+func watchdogConfPath() string {
+	return filepath.Join(dirs.GlobalRootDir, "/etc/systemd/system.conf.d/")
+}
+
+func watchdogConfEnvironment() string {
+	return filepath.Join(watchdogConfPath(), "10-ubuntu-core-watchdog.conf")
+}
+
+func updateWatchdogConfig(config map[string]uint) error {
+	path := watchdogConfEnvironment()
+
+	configStr := ""
+	for k, v := range config {
+		if v > 0 {
+			configStr += fmt.Sprintf("%s=%d\n", k, v)
+		}
+	}
+	if len(configStr) == 0 {
+		// No config, remove file. An error would indicate that the file
+		// does not exist anyway, so we ignore it.
+		os.Remove(path)
```

Comment:
Maybe `if err := os.Remove(path); err != nil && !os.IsNotExist(err) { return err }` ? This way we catch the other potential errors that unlink may produce.